### DISCUSSION TITLE
What's the best way to execute a script once, rather than going into an event loop?

### DISCUSSION THREAD

**Discussion by jammur** [Rapptz/discord.py#5896] [Q&A] [answered]
I need to run a script on a schedule that will print a few messages to a specific channel. Is there a way to do this without getting into an event loop and having to exit the program with exit(), quit(), etc when done?

**Comment by Rapptz** [comment, marked as answer]
You should probably use a [webhook](https://support.discord.com/hc/en-us/articles/228383668-Intro-to-Webhooks).

[The library supports webhooks](https://discordpy.readthedocs.io/en/latest/api.html#webhook-support) in both async (`aiohttp`) and sync (`requests`) mode so once you get your webhook URL you can use the `requests` library and post to it. For example:

```py
from discord import Webhook, RequestsWebhookAdapter

webhook_url = 'your webhook URL goes here'
webhook = Webhook.from_url(webhook_url, adapter=RequestsWebhookAdapter())

# Send a message to the channel
webhook.send('Hello world!')
```

**Reply by jammur** [reply]
@Rapptz This is exactly what I needed. Thanks!
